client/src/components/Videos.js

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+import staticData from "../exampleresponse.json";
+import AddVideoForm from "./AddVideoForm";
+import Search from "./Search";
+import SingleVideo from "./SingleVideo";
+import "./Videos.css";
+
+const Videos = () => {
+  const sortedVideosData = staticData.sort((a, b) => b.rating - a.rating);
+  const [videosData, setVideosData] = useState(sortedVideosData);
+  const [searchValue, setSearchValue] = useState("");
+  const [showError, setShowError] = useState(false);
+
+  // delete video (for SingleVideo)
+  const handleDeleteVideoClick = (id) => {
+    const newVideosData = videosData.filter((video) => {
+      return video.id !== id;
+    });
+    setVideosData(newVideosData);
+  };
+
+  const handleSearch = (event) => {
+    setSearchValue(event.target.value);
+  };
+
+  // matchYoutubeUrl was copied from here: https://www.codegrepper.com/code-examples/javascript/javascript+validate+url+to+match+youtube+video
+  const matchYoutubeUrl = (url) => {
+    var p =
+      /^(?:https?:\/\/)?(?:m\.|www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
+    if (url.match(p)) {
+      return url.match(p)[1];
+    }
+    return false;
+  };
+
+  const handleAddVideo = (title, url) => {
+    if (title !== "" && url !== "" && matchYoutubeUrl(url)) {
+      const newVideoObj = {
+        id: Date.now(),
+        title,
+        url,
+        rating: 0,
+      };
+
+      const newVideosData = [...videosData]; // creates copy of videosData
+      newVideosData.push(newVideoObj);
+      setVideosData(newVideosData);
```

Comment:
It's good that you're using spread here. Perhaps a neater way of doing it would be to say:
`setVideosData([...videosData, newVideoObj])`
It's equivalent to what you have here, just a little more concise.